team/bundles/org.eclipse.compare/compare/org/eclipse/compare/contentmergeviewer/TextMergeViewer.java

```diff
@@ -5408,27 +5408,32 @@ private boolean isIgnoreAncestor() {
 
 	private void updateToolbarLabel() {
 		final String DIFF_COUNT_ID = "DiffCount"; //$NON-NLS-1$
+		boolean isUpdateNeeded = false;
 		ToolBarManager tbm =
 				(ToolBarManager) getToolBarManager(fComposite.getParent());
 		int differenceCount = fMerger.changesCount();
-		if (tbm != null) {
+		if (tbm != null && tbm.getItems().length > 0) {
 
 			String label = MessageFormat.format(CompareMessages.TextMergeViewer_differences, differenceCount);
 			LabelContributionItem labelContributionItem = new LabelContributionItem(DIFF_COUNT_ID,
 					label);
 
 			if (tbm.find(DIFF_COUNT_ID) != null) {
 				tbm.replaceItem(DIFF_COUNT_ID, labelContributionItem);
-			} else {
+				isUpdateNeeded = true;
+			} else if (tbm.find("diffLabel") != null) { //$NON-NLS-1$
 				tbm.appendToGroup("diffLabel", labelContributionItem); //$NON-NLS-1$
+				isUpdateNeeded = true;
 			}
+			if (isUpdateNeeded) {
 			fComposite.getDisplay().asyncExec(() -> {
 				// relayout in next tick
 				ToolBar control = tbm.getControl();
 				if (control != null && !control.isDisposed()) {
 					tbm.update(true);
 				}
 			});
```

Comment:
Could you please correct the indentation here?

Reply:
The second commit had added a newline at the end of the method and also unnecessary tabs in an empty line inside the method. 
I deleted the empty line and formatted (`Ctrl + Shift + F`) the whole method. I pushed the changes so you don't need to do anything else :-)

Reply:
Thank you so much !

Reply:
> The second commit had added a newline at the end of the method and also unnecessary tabs in an empty line inside the method. I deleted the empty line and formatted (`Ctrl + Shift + F`) the whole method. I pushed the changes so you don't need to do anything else :-)

Have you done the changes related to paintlistener - changing the code to use lambda expressions?

Reply:
Those changes and the change from `while`-loop to `for`-loop in `disposeCompareFilterActions` were done by save actions. They're fine by me so I think they should stay. Unless you want me to revert them?

Reply:
Oh! That's great.. Just wanted to know If I committed something by mistake.

Reply:
No no, that was all me, you did nothing wrong :-)

You can still see your original commit BTW. You can find it in the information below ;-)

![image](https://github.com/eclipse-platform/eclipse.platform/assets/2205684/cb2dd7b8-4e9b-4f13-a190-9803dcffa520)

Reply:
> Those changes and the change from `while`-loop to `for`-loop in `disposeCompareFilterActions` were done by save actions.

I personally would prefer to separate small fix from huge number of unrelated cleanups. The reason is simple - if one needs to understand later why something is changed, one will see commit with fully unrelated explanation and code...

Reply:
Then I'll separate them and merge this PR right after the checks ran.

Would that be ok for you (two)?

Reply:
Yes, please.

Reply:
Done: https://github.com/eclipse-platform/eclipse.platform/pull/1320